test/hotspot/jtreg/runtime/8339725/libagent8339725.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <jvmti.h>
+#include <jni.h>
+#include <pthread.h>
+
+static jvmtiEnv *jvmti;
+static JavaVM *_jvm;
+static JNIEnv *_rb_env;
+
+#define BUFFER_SIZE 100000
+static size_t ring_buffer[BUFFER_SIZE] = {0};
+static volatile int ring_buffer_idx = 0;
+static int reader_created = 0;
+
+void *get_method_details(void *arg)
+{
+    jmethodID method = (jmethodID)arg;
+
+    jclass method_class;
+    char *class_name = NULL;
+
+    jvmtiError err = JVMTI_ERROR_NONE;
+
+    // For JVM 17, 21, 22 calling GetMethodDeclaringClass is enough.
+    if ((err = jvmti->GetMethodDeclaringClass(method, &method_class)) == 0)
+    {
+        // JVM 8 needs this to crash
+        jvmti->GetClassSignature(method_class, &class_name, NULL);
```

Comment:
The good practice is to check jvmti status and fails if it is not none.
Check the 'check_jvmti_status' for this from 'jvmti_common.h"
Please check all jvmti functions.

Reply:
Added result check for all jvmti calls except `Deallocate`.